src/client/apps/edit/components/header/index.tsx

```diff
@@ -0,0 +1,284 @@
+import { changeView } from "../../../../../client/actions/edit/editActions"
+
+interface Props {
+  article: ArticleData
+  beforeUnload: () => void
+  changeViewAction: (e) => void
+  channel: Channel
+  deleteArticleAction: () => void
+  edit: Edit
+  forceURL: string
+  isAdmin: boolean
+  publishArticleAction: () => void
+  saveArticleAction: () => void
+}
+
+interface Channel {
+  id: string
+  name: string
+  type: string
+}
+
+interface Edit {
+  activeView: string
+  article: ArticleData
+  currentSession: object
+  error: object
+  isDeleting: boolean
+  isPublishing: boolean
+  isSaved: true
+  isSaving: false
+  mentioned: object
+  section: object
+  sectionIndex: number
+  setYoastKeyword: string
+}
+
+export class EditHeader extends Component<Props> {
+  isPublishable = () => {
+    return this.finishedContent() && this.finishedDisplay()
+  }
+
+  finishedContent = () => {
+    const { title } = this.props.article
+
+    return title && title.length > 0
+  }
+
+  finishedDisplay = () => {
+    const { thumbnail_image, thumbnail_title } = this.props.article
+    const finishedImg = thumbnail_image && thumbnail_image.length > 0
+    const finishedTitle = thumbnail_title && thumbnail_title.length > 0
+
+    return finishedImg && finishedTitle
+  }
+
+  onPublish = () => {
+    const { publishArticleAction } = this.props
+
+    if (this.isPublishable()) {
+      publishArticleAction()
+    }
+  }
+
+  onSave = () => {
+    const { saveArticleAction } = this.props
+
+    this.removeUnsavedAlert()
+    saveArticleAction()
+  }
+
+  onDelete = () => {
+    const { deleteArticleAction } = this.props
+
+    if (confirm("Are you sure?")) {
+      this.removeUnsavedAlert()
+      deleteArticleAction()
+    }
+  }
+
+  removeUnsavedAlert = () => {
+    const { beforeUnload } = this.props
+    // dont show popup for unsaved changes when saving/deleting
+    window.removeEventListener("beforeunload", beforeUnload)
+  }
+
+  getSaveColor = () => {
+    const { isSaving, isSaved } = this.props.edit
+
+    if (isSaving) {
+      return "green100"
+    } else if (isSaved) {
+      return "black100"
+    } else {
+      return "red100"
+    }
+  }
+
+  getSaveText = () => {
+    const { article, edit } = this.props
+    const { isSaving } = edit
+
+    if (isSaving) {
+      return "Saving..."
+    } else if (article.published) {
+      return "Save Article"
+    } else {
+      return "Save Draft"
+    }
+  }
+
+  getPublishText = () => {
+    const { article, edit } = this.props
+    const { isPublishing } = edit
+    const isPublished = article.published
+
+    if (isPublishing && isPublished) {
+      return "Unpublishing..."
+    } else if (isPublishing) {
+      return "Publishing..."
+    } else if (isPublished) {
+      return "Unpublish"
+    } else {
+      return "Publish"
+    }
+  }
+  handleTabChange = tab => {
+    this.props.changeViewAction(tab)
+  }
+
+  addTabIcon = tab => {
+    if (tab.props.children === "Content") {
```

Comment:
It looks like the `Tab` component accepts children, curious why you chose to clone the element rather than wrapping the icon inside the component, was this not working?